### PR TITLE
New desktop header

### PR DESCRIPTION
**Pull request by jenibelle08**
### Description

This PR is an initial version of the desktop menu for the homepage revamp. The mobile menu is temporarily removed, so that the desktop menu can be reviewed in isolation. 

*This PR is not intended for merge into master until the completion of the homepage revamp*

### Notes

There is still some refactoring and cleanup to do (particularly in the CSS) that will be done as part of a subsequent larger PR that includes the mobile menu. However, the functionality of the desktop can be previewed.

### CODE REVIEW

File: static/js/apps/base/components/menu_desktop_rich_menu.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  open: boolean;
+}
+
+const MenuDesktopRichMenu = ({
+  menuItem,
+  routes,
+  open,
+}: MenuDesktopRichMenuProps): ReactElement => {
+  return (
+    <div className={`rich-menu-content ${open ? "open" : ""}`}>
```
Comment (beets) [inline static/js/apps/base/components/menu_desktop_rich_menu.tsx:39]: this looks like the only difference between desktop and mobile -- can we fold both into the same component (ok to add a todo for now).

**Review by beets — APPROVED** [review]
thanks for this - it's nice to see it working across the site. i added some comments - they are mostly very minor and have to do with naming. could you add them as todo's and we can merge this.

File: server/config/base/header_v2.json [hunk 2]
```diff
@@ -0,0 +1,203 @@
+[
```
Comment (dwnoble) [inline server/config/base/header_v2.json:1]: We'll have to think about a good way to add translations to these strings (in a follow-up pr)

File: static/js/apps/base/components/header_bar/menu_desktop_rich_menu.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
+/**
```
Comment (dwnoble) [inline static/js/apps/base/components/header_bar/menu_desktop_rich_menu.tsx:1]: (For a follow-up PR): When scrolling down the page with the menu open, the menu sticks to the top.

**Review by dwnoble — APPROVED** [review]
Looks great!

**Pull request merged**